### PR TITLE
remove delete org and create a new location for virt-who upgrade scenarios

### PR DESCRIPTION
**Pull request by yanpliu**
remove delete org and create a new location for virt-who upgrade scenarios
Cases : PASS
```robottelo_vv) [virtwho@dell-per740-68-vm-04 robottelo]$ pytest ./tests/upgrades/test_virtwho.py  -m pre_upgrade 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.4.0, pluggy-1.0.0
Mandatory Requirements are up to date.
Optional Requirements are up to date.
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/virtwho/repository/robottelo
configfile: pyproject.toml
plugins: cov-4.1.0, ibutsu-2.2.4, xdist-3.3.1, mock-3.11.1, services-2.2.1, reportportal-5.2.0
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_virtwho.py .                                                                                                                                                                            [100%]

=================================================================================== 1 passed, 1 deselected in 124.62s (0:02:04) ===================================================================================
(robottelo_vv) [virtwho@dell-per740-68-vm-04 robottelo]$ pytest ./tests/upgrades/test_virtwho.py  -m post_upgrade 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.4.0, pluggy-1.0.0
Mandatory Requirements are up to date.
Optional Requirements are up to date.
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/virtwho/repository/robottelo
configfile: pyproject.toml
plugins: cov-4.1.0, ibutsu-2.2.4, xdist-3.3.1, mock-3.11.1, services-2.2.1, reportportal-5.2.0
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_virtwho.py .                                                                                                                                                                            [100%]

======================================================================================== 1 passed, 1 deselected in 20.05s =========================================================================================
```

### CODE REVIEW

**Review by adarshdubey-star — APPROVED** [review]

**Comment by adarshdubey-star** [comment]
Tagging Tier-2 reviewers,  as I am not sure how the pre/post upgrade test should be tested via PRT as we need it for cherrypicking.

**Review by shweta83 — APPROVED** [review]

File: tests/upgrades/test_virtwho.py [hunk 1]
```diff
@@ -78,16 +77,8 @@ def test_pre_create_virt_who_configuration(
...
-        if org:
-            target_sat.api.Organization(id=org[0].id).delete()
-        default_loc_id = (
-            target_sat.api.Location().search(query={'search': f'name="{DEFAULT_LOC}"'})[0].id
-        )
-        default_loc = target_sat.api.Location(id=default_loc_id).read()
         org = target_sat.api.Organization(name=ORG_DATA['name']).create()
-        default_loc.organization.append(target_sat.api.Organization(id=org.id))
-        default_loc.update(['organization'])
+        target_sat.api.Location(organization=[org]).create()
```
Comment (Gauravtalreja1) [inline tests/upgrades/test_virtwho.py:81]: this location created isn't assigned to any variable, so question is if you really need it in test for pre/post scenarios?

Comment (yanpliu) [reply tests/upgrades/test_virtwho.py:81]: > this location created isn't assigned to any variable, so question is if you really need it in test for pre/post scenarios?

The location is assigned to the org "org = target_sat.api.Organization(name=ORG_DATA['name']).create()"

Comment (Gauravtalreja1) [reply tests/upgrades/test_virtwho.py:81]: yes, it is created with org, but there seems no use of it, right?

**Review by synkd — APPROVED** [review]

File: tests/upgrades/test_virtwho.py (hunk 1 above)
Comment (yanpliu) [reply tests/upgrades/test_virtwho.py:81]: > yes, it is created with org, but there seems no use of it, right?

@Gauravtalreja1 This location is used for org = target_sat.api.Organization(name=ORG_DATA['name']).create(),  then create a virtwho config in this org

**Review by Gauravtalreja1 — APPROVED** [review]

File: tests/upgrades/test_virtwho.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/upgrades/test_virtwho.py:81]: Np, it is non-blocking, so going for merge

**Pull request merged**